lib/auth/auth.go

```diff
@@ -2948,6 +2952,10 @@ func (a *Server) CreateAuthenticateChallenge(ctx context.Context, req *proto.Cre
 			return nil, trace.Wrap(err)
 		}
 
+		if challengeExtensions.Scope == mfav1.ChallengeScope_CHALLENGE_SCOPE_UNSPECIFIED {
+			challengeExtensions.Scope = mfav1.ChallengeScope_CHALLENGE_SCOPE_LOGIN
+		}
```

Comment:
I'm unclear whether this is checked elsewhere, but this might include a sanity check like:

```go
if challengeExtensions.Scope != mfav1.ChallengeScope_CHALLENGE_SCOPE_LOGIN {
	if challengeExtensions.Scope == mfav1.ChallengeScope_CHALLENGE_SCOPE_UNSPECIFIED {
		challengeExtensions.Scope = mfav1.ChallengeScope_CHALLENGE_SCOPE_LOGIN
	} else {
		return nil, trace.BadParameter("invalid scope %q for password authentication", challengeExtensions.Scope)
	}
}
```

Because there are similar ones below, you could create this function

```go
func validateAndSetScope(challengeExtensions *mfav1.ChallengeExtensions, expectedScope mfav1.ChallengeScope, context string) error {
	if challengeExtensions.Scope != expectedScope {
		if challengeExtensions.Scope == mfav1.ChallengeScope_CHALLENGE_SCOPE_UNSPECIFIED {
			challengeExtensions.Scope = expectedScope
		} else {
			return trace.BadParameter("invalid scope %q for %s", challengeExtensions.Scope, context)
		}
	}
	return nil
}
```

and then use it like 

```go
if err := validateAndSetScope(challengeExtensions, mfav1.ChallengeScope_CHALLENGE_SCOPE_LOGIN, "password authentication"); err != nil {
	return nil, err
}
```

```go
if err := validateAndSetScope(challengeExtensions, mfav1.ChallengeScope_CHALLENGE_SCOPE_ACCOUNT_RECOVERY, "account recovery"); err != nil {
	return nil, err
}
```

```go
if err := validateAndSetScope(challengeExtensions, mfav1.ChallengeScope_CHALLENGE_SCOPE_PASSWORDLESS_LOGIN, "passwordless authentication"); err != nil {
	return nil, err
}
```

(take it or leave it).